src/citrine/seeding/parse_csv.py

```diff
@@ -0,0 +1,450 @@
+            process=ProcessRun(name=row[process_header]),
+        )
+        mat_run.process.spec = ProcessSpec(
+            name=mat_run.process.name, template=obj_tmpls[mat_run.process.name],
+        )
+        mat_run.spec = MaterialSpec(
+            name=mat_run.name, process=mat_run.process.spec, template=obj_tmpls[row[class_header]],
+        )
+
+        if mat_name in mat_index:
+            raise ValueError("Repeat material identifier: {}".format(mat_name))
+        mat_index[mat_name] = mat_run
+
+        links[mat_name] = []
+        for i in range(max_id + 1):
+            ing_name = ing_name_tmpl.format(i)
+            ing_amt = ing_amt_tmpl.format(i)
+            if pd.notnull(row[ing_name]):
+                links[mat_name].append((row[ing_name], row[ing_amt]))
+
+        row_msr = OrderedDict()
+        for key in row.dropna().keys():
+            if key not in val_builder:
+                continue
+            attr = val_builder[key](row[key])
+
+            # Simplify here, since only certain types end up in certain places
+            if isinstance(attr, Property):
+                msr_name = tmpl_links[key]
+                if msr_name not in row_msr:
+                    msr_spec = MeasurementSpec(name=msr_name, template=obj_tmpls[msr_name])
+                    msr_run = MeasurementRun(name=msr_name, material=mat_run, spec=msr_spec)
+                    row_msr[msr_name] = msr_run
+                else:
+                    msr_run = row_msr[msr_name]
+                msr_run.properties.append(attr)
+            elif isinstance(attr, Condition):
+                mat_run.process.conditions.append(attr)
+            else:
+                raise TypeError("Unhandled type: {}".format(type(attr)))
+
+    for downstream in links:
+        if fraction_basis == FractionBasis.NOT_NORMALIZED:
+            total = sum(x[1] for x in links[downstream])
+            if len(links[downstream]) > 0 and total <= 0.0:
+                raise ValueError(f"Sum of ingredients to {downstream} was {total}")
+        else:
+            total = fraction_basis.value
+
+        for i, (upstream, amount) in enumerate(links[downstream]):
+            ing_name = "Subsystem {}".format(i)
+            label_list = df.loc[df[name_header] == upstream][[class_header]].values.reshape(-1)
+
+            ing_spec = IngredientSpec(
+                name=ing_name,
+                material=mat_index[upstream].spec,
+                process=mat_index[downstream].spec.process,
+                labels=list(label_list),
+            )
+
+            # TODO: Option for mass/number/volume/absolute
+            ing_fraction = NominalReal(nominal=amount / total, units="")
+            ing_run = IngredientRun(  # noqa: F841  Easier to follow w/ unused variable
+                material=mat_index[upstream],
+                spec=ing_spec,
+                process=mat_index[downstream].process,
+                mass_fraction=ing_fraction,
```

Comment:
It would be nice to have an option to choose which quantity the columns are indicating. Is this something that should be defined in a header annotation? Or a method argument on the builder?